src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp

```diff
@@ -33,27 +33,69 @@ bool DiscoveryDataBase::pdp_is_relevant(
         const eprosima::fastrtps::rtps::CacheChange_t& change,
         const eprosima::fastrtps::rtps::GUID_t& reader_guid) const
 {
-    (void)change;
-    (void)reader_guid;
-    return true;
+    auto it = participants_.find(guid_from_change(&change).guidPrefix);
```

Comment:
In general, don't we need to lock the mutex before any access to the data structures?